apps/src/lib/levelbuilder/lesson-editor/AddResourceDialog.jsx

```diff
@@ -18,33 +60,212 @@ const styles = {
     isOpen: PropTypes.bool.isRequired,
-    handleConfirm: PropTypes.func.isRequired
+    onSave: PropTypes.func,
+    handleClose: PropTypes.func.isRequired,
+    typeOptions: PropTypes.arrayOf(PropTypes.string),
+    audienceOptions: PropTypes.arrayOf(PropTypes.string)
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      key: '',
+      name: '',
+      type: '',
+      audience: '',
+      pdf: false,
+      assessment: false,
+      url: '',
+      downloadUrl: '',
+      error: ''
+    };
+  }
+
+  validateResource = () => {
+    const {key, name, url} = this.state;
+    let error = '';
+    if (name === '') {
+      error += 'Name is required. ';
+    }
+    if (url === '') {
+      error += 'URL is required. ';
+    }
+    if (key === '') {
+      error += 'Embed slug is required.';
+    }
+    if (error === '') {
+      return true;
+    } else {
+      this.setState({error});
+      return false;
+    }
+  };
+
+  resetState = () => {
+    this.setState({
+      key: '',
+      name: '',
+      type: '',
+      audience: '',
+      pdf: false,
+      assessment: false,
+      url: '',
+      downloadUrl: '',
+      error: ''
+    });
+  };
+
+  saveResource = e => {
+    e.preventDefault();
+    if (this.validateResource()) {
+      const formData = new FormData(e.target);
+      fetch('/resources', {
+        method: 'POST',
+        headers: {'X-CSRF-Token': formData.get('authenticity_token')},
+        body: formData
+      })
+        .then(response => (response.ok ? response.json() : {}))
+        .then(json => {
+          if (json !== {}) {
+            this.resetState();
+            if (this.props.onSave) {
+              this.props.onSave(json);
+            }
+            this.props.handleClose();
+          }
+        })
+        .catch(error => this.setState({error}));
+    }
+  };
+
+  handleInputChange = e => {
+    const target = e.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    this.setState({[target.name]: value});
   };
 
   render() {
     return (
       <BaseDialog
         isOpen={this.props.isOpen}
-        handleClose={this.props.handleConfirm}
+        handleClose={this.props.handleClose}
         useUpdatedStyles
         style={styles.dialog}
       >
         <h2>Add Resource</h2>
-        <label>
-          <span>Add link to resource:</span>
-          <select onChange={() => {}}>
-            <option>Slides</option>
-            <option>Activity Guide</option>
-            <option>Video</option>
-          </select>
-        </label>
-        <DialogFooter rightAlign>
-          <Button
-            text={'Close and Add'}
-            onClick={this.props.handleConfirm}
-            color={Button.ButtonColor.orange}
-          />
-        </DialogFooter>
+        {this.state.error !== '' && (
+          <h3 style={{color: 'red'}}>{this.state.error}</h3>
+        )}
+        <form id="create-resource-form" onSubmit={this.saveResource}>
```

Comment:
Why create a form inside this dialog? Feels like that is adding complication just as the rail authenticity token

Reply:
interesting, I think I've always favored doing this with ajax. I don't see a specific problem with doing it this way, though. If I am reading it correctly the only added hoops are the RailsAuthenticityToken and the call to preventDefault.

Reply:
I don't even think Ajax would be needed here because the resource isn't being created until the lesson is saved. So maybe its can be simplified?

Reply:
This dialog does create the resource. There's a part in the spec about adding the resource to other lessons via this dialog, which I didn't add in this PR, that seems like it would create the expectation that the resource is saved. Open to thoughts on this though!

Reply:
@bethanyaconnor  I think this sounds right. otherwise I don't see how you would be able to find the resource you just created in the dropdown.

Reply:
Ah ok sorry I got confused. Thanks for sharing another way to set these things up!